joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/JavascriptCallLinker.scala

```diff
@@ -156,9 +156,19 @@ class JavascriptCallLinker(cpg: Cpg) extends CpgPass(cpg) {
       .nonEmpty
   }
 
-  private def fromFieldAccess(c: Call): Option[String] = c.methodFullName match {
-    case Operators.fieldAccess if isArgumentFromExport(c) || isFieldAccessWithThisBase(c) => Option(c.argument(2).code)
-    case _                                                                                => None
+  private def isFieldAccessWithType(call: Call): Boolean = {
+    // if the type of argument = 1 of the field access is known and a matching type decl member can be found
+    // TODO: make this work for for access chains, e.g., "a.b.c.foo()"
+    val fieldName = call.argument(2).code
+    val typeDeclCandidates =
+      call.argument(1).collectAll[Identifier].dynamicTypeHintFullName.flatMap(cpg.typeDecl.nameExact)
```

Comment:
If we start using these in joern, we should really move the `DynamicTypeLinker` from CS to joern instead of adding (incompatible) ad-hoc queries like this one.

Also if we resolve this to a specific type here, then we should also return that type here - this is currently running in the context of https://github.com/joernio/joern/blob/f6d00c3a060864692f7146e1f557226382c678af/joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/JavascriptCallLinker.scala#L95 which doesn't really make sense in a case where the type is already known.